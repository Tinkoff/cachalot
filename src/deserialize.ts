import * as errors from "./errors/errors";

export default function <R>(value: string): R {
  try {
    return JSON.parse(value);
  } catch (error) {
    throw errors.parseError(error);
  }
}
