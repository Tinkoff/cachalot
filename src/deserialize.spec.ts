import deserialize from "./deserialize";

describe("deserizalize", () => {
  it("parses valid json string or null", () => {
    expect(deserialize("null")).toEqual(null);
    expect(deserialize('"test"')).toEqual("test");
    expect(deserialize("1")).toEqual(1);
    expect(deserialize('{"a":1}')).toEqual({ a: 1 });
  });

  it("throws parse error on invalid json string", () => {
    expect(() => deserialize("test")).toThrowErrorMatchingSnapshot();
    expect(() => deserialize('{"a":')).toThrowErrorMatchingSnapshot();
  });
});
