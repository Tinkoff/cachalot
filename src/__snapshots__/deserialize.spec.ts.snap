// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deserizalize throws parse error on invalid json string 1`] = `"Unexpected token e in JSON at position 1"`;

exports[`deserizalize throws parse error on invalid json string 2`] = `"Unexpected end of JSON input"`;
